test/integration/scheduler/preemption_test.go

```diff
@@ -932,98 +933,209 @@ func TestPreemptionRaces(t *testing.T) {
+					st.MakePod().Name("low-4").Priority(lowPriority).Req(map[v1.ResourceName]string{v1.ResourceCPU: "1"}).Obj(),
+				},
+				{
+					st.MakePod().Name("medium").Priority(mediumPriority).Req(map[v1.ResourceName]string{v1.ResourceCPU: "4"}).Obj(),
+				},
+				{
+					st.MakePod().Name("high").Priority(highPriority).Req(map[v1.ResourceName]string{v1.ResourceCPU: "3"}).Obj(),
+				},
+			},
+			postChecks: []func(cs clientset.Interface, pod *v1.Pod) error{
+				testutils.WaitForPodToSchedule,
+				waitForNominatedNodeName,
+				waitForNominatedNodeName,
+			},
 		},
-	})
-	medPriPod, err := createPausePod(cs, podConf)
-	if err != nil {
-		t.Errorf("Error while creating the medium priority pod: %v", err)
-	}
-	// Step 3. Check if .status.nominatedNodeName of the medium priority pod is set.
-	if err := waitForNominatedNodeName(cs, medPriPod); err != nil {
-		t.Errorf(".status.nominatedNodeName was not set for pod %v/%v: %v", medPriPod.Namespace, medPriPod.Name, err)
-	}
-	// Step 4. Create a high priority pod.
-	podConf = initPausePod(&pausePodConfig{
-		Name:      "high-priority",
-		Namespace: testCtx.NS.Name,
-		Priority:  &highPriority,
-		Resources: &v1.ResourceRequirements{Requests: v1.ResourceList{
-			v1.ResourceCPU:    *resource.NewMilliQuantity(300, resource.DecimalSI),
-			v1.ResourceMemory: *resource.NewQuantity(200, resource.DecimalSI)},
+		{
+			name:         "mid-priority pod preempts low-priority pod, followed by a high-priority pod without additional preemption",
+			nodeCapacity: map[v1.ResourceName]string{v1.ResourceCPU: "2"},
+			podsToCreate: [][]*v1.Pod{
+				{
+					st.MakePod().Name("low").Priority(lowPriority).Req(map[v1.ResourceName]string{v1.ResourceCPU: "1"}).Obj(),
+				},
+				{
+					st.MakePod().Name("medium").Priority(mediumPriority).Req(map[v1.ResourceName]string{v1.ResourceCPU: "2"}).Obj(),
+				},
+				{
+					st.MakePod().Name("high").Priority(highPriority).Req(map[v1.ResourceName]string{v1.ResourceCPU: "1"}).Obj(),
+				},
+			},
+			postChecks: []func(cs clientset.Interface, pod *v1.Pod) error{
+				testutils.WaitForPodToSchedule,
+				waitForNominatedNodeName,
+				testutils.WaitForPodToSchedule,
+			},
+			podNamesToDelete: []string{"low"},
+		},
+		{
+			name:         "mid-priority pod preempts low-priority pod, followed by a node deletion",
+			nodeCapacity: map[v1.ResourceName]string{v1.ResourceCPU: "1"},
+			podsToCreate: [][]*v1.Pod{
+				{
+					st.MakePod().Name("low").Priority(lowPriority).Req(map[v1.ResourceName]string{v1.ResourceCPU: "1"}).Obj(),
+				},
+				{
+					st.MakePod().Name("medium").Priority(mediumPriority).Req(map[v1.ResourceName]string{v1.ResourceCPU: "1"}).Obj(),
+				},
+			},
+			postChecks: []func(cs clientset.Interface, pod *v1.Pod) error{
+				testutils.WaitForPodToSchedule,
+				waitForNominatedNodeName,
+			},
+			// Delete the node to simulate an ErrNoNodesAvailable error.
+			deleteNode:       true,
+			podNamesToDelete: []string{"low"},
```

Comment:
In a real cluster, this is actually optional.

However, in an integration test it may flake without this - a node deletion event wouldn't trigger moving the preemptor back to activeQ, so if the preemptor happens to be backoffed and the 30 seconds timeout may not be enough. So add a pod deletion here to actively trigger retrying the preemptor.